// @okgrow/graphql-scalars/dist/scalars.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.URLScalar = exports.UnsignedIntScalar = exports.UnsignedFloatScalar = exports.RegularExpressionScalar = exports.PostalCodeScalar = exports.PositiveIntScalar = exports.PositiveFloatScalar = exports.PhoneNumberScalar = exports.NonPositiveIntScalar = exports.NonPositiveFloatScalar = exports.NonNegativeIntScalar = exports.NonNegativeFloatScalar = exports.NegativeIntScalar = exports.NegativeFloatScalar = exports.EmailAddressScalar = exports.DateTimeScalar = void 0;
var DateTimeScalar = 'scalar DateTime';
exports.DateTimeScalar = DateTimeScalar;
var EmailAddressScalar = 'scalar EmailAddress';
exports.EmailAddressScalar = EmailAddressScalar;
var NegativeFloatScalar = 'scalar NegativeFloat';
exports.NegativeFloatScalar = NegativeFloatScalar;
var NegativeIntScalar = 'scalar NegativeInt';
exports.NegativeIntScalar = NegativeIntScalar;
var NonNegativeFloatScalar = 'scalar NonNegativeFloat';
exports.NonNegativeFloatScalar = NonNegativeFloatScalar;
var NonNegativeIntScalar = 'scalar NonNegativeInt';
exports.NonNegativeIntScalar = NonNegativeIntScalar;
var NonPositiveFloatScalar = 'scalar NonPositiveFloat';
exports.NonPositiveFloatScalar = NonPositiveFloatScalar;
var NonPositiveIntScalar = 'scalar NonPositiveInt';
exports.NonPositiveIntScalar = NonPositiveIntScalar;
var PhoneNumberScalar = 'scalar PhoneNumber';
exports.PhoneNumberScalar = PhoneNumberScalar;
var PositiveFloatScalar = 'scalar PositiveFloat';
exports.PositiveFloatScalar = PositiveFloatScalar;
var PositiveIntScalar = 'scalar PositiveInt';
exports.PositiveIntScalar = PositiveIntScalar;
var PostalCodeScalar = 'scalar PostalCode';
exports.PostalCodeScalar = PostalCodeScalar;
var RegularExpressionScalar = 'scalar RegularExpression';
exports.RegularExpressionScalar = RegularExpressionScalar;
var UnsignedFloatScalar = 'scalar UnsignedFloat';
exports.UnsignedFloatScalar = UnsignedFloatScalar;
var UnsignedIntScalar = 'scalar UnsignedInt';
exports.UnsignedIntScalar = UnsignedIntScalar;
var URLScalar = 'scalar URL';
exports.URLScalar = URLScalar;
var _default = [DateTimeScalar, EmailAddressScalar, NegativeFloatScalar, NegativeIntScalar, NonNegativeFloatScalar, NonNegativeIntScalar, NonPositiveFloatScalar, NonPositiveIntScalar, PhoneNumberScalar, PositiveFloatScalar, PositiveIntScalar, PostalCodeScalar, RegularExpressionScalar, UnsignedFloatScalar, UnsignedIntScalar, URLScalar];
exports.default = _default;
